src/MainView/include/MainView/MainView.h

```diff
@@ -1,11 +1,15 @@
 #pragma once
 
-#include <QWidget>
+#include <QHBoxLayout>
```

Comment:
It seems we can forward declare `QHBoxLayout` and `QPushButton` here.
`QWidget` is needed as `MainView` is derived from it, however we only have pointers for other classes, so we can forward declare them in header file and include in cpp file. 
This is a nice blog post on this topic https://www.incredibuild.com/blog/optimizing-cpp-header-files